src/binding/c/attr_api.txt

```diff
@@ -0,0 +1,202 @@
+    .seealso: MPI_Attr_put, MPI_Keyval_create, MPI_Attr_delete, MPI_Comm_get_attr
+/*
+    Notes:
+        Attributes must be extracted from the same language as they were inserted
+        in with 'MPI_ATTR_PUT'.  The notes for C and Fortran below explain why.
+*/
+/*
+    Notes for C:
+        Even though the 'attribute_val' argument is declared as 'void *', it is
+        really the address of a void pointer (i.e., a 'void **').  Using
+        a 'void *', however, is more in keeping with C idiom and allows the
+        pointer to be passed without additional casts.
+
+    .N Fortran
+        The 'attribute_val' in Fortran is a pointer to a Fortran integer, not
+        a pointer to a 'void *'.
+*/
+
+MPI_Attr_put:
+    .desc: Stores attribute value associated with a key
+    .replace: deprecated with MPI_Comm_set_attr
+    .seealso: MPI_Attr_get, MPI_Keyval_create, MPI_Attr_delete, MPI_Comm_set_attr
+/*
+    Notes:
+    Values of the permanent attributes 'MPI_TAG_UB', 'MPI_HOST', 'MPI_IO',
+    'MPI_WTIME_IS_GLOBAL', 'MPI_UNIVERSE_SIZE', 'MPI_LASTUSEDCODE', and
+    'MPI_APPNUM'  may not be changed.
+
+    The type of the attribute value depends on whether C, C++, or Fortran
+    is being used.
+    In C and C++, an attribute value is a pointer ('void *'); in Fortran,
+    it is a single
+    integer (`not` a pointer, since Fortran has no pointers and there are systems
+    for which a pointer does not fit in an integer (e.g., any > 32 bit address
+    system that uses 64 bits for Fortran 'DOUBLE PRECISION').
+
+    If an attribute is already present, the delete function (specified when the
+    corresponding keyval was created) will be called.
+
+*/
+
+MPI_Comm_create_keyval:
+    .desc: Create a new attribute key
+    .seealso: MPI_Comm_free_keyval
+/*
+    Notes:
+    Key values are global (available for any and all communicators).
+
+    Default copy and delete functions are available.  These are
+    + MPI_COMM_NULL_COPY_FN   - empty copy function
+    . MPI_COMM_NULL_DELETE_FN - empty delete function
+    - MPI_COMM_DUP_FN         - simple dup function
+
+    There are subtle differences between C and Fortran that require that the
+    copy_fn be written in the same language from which 'MPI_Comm_create_keyval'
+    is called.
+    This should not be a problem for most users; only programmers using both
+    Fortran and C in the same program need to be sure that they follow this rule.
+
+    .N AttrErrReturn
+*/
+
+MPI_Comm_delete_attr:
+    .desc: Deletes an attribute value associated with a key on
+    .seealso: MPI_Comm_set_attr, MPI_Comm_create_keyval
+    .extra: ignore_revoke
+
+MPI_Comm_free_keyval:
+    .desc: Frees an attribute key for communicators
+{
+    MPIR_free_keyval(comm_keyval_ptr);
+    *comm_keyval = MPI_KEYVAL_INVALID;
+}
+
+MPI_Comm_get_attr:
+    .desc: Retrieves attribute value by key
+    .polymorph: MPIR_Attr_type attr_type=MPIR_ATTR_PTR
+    .extra: ignore_revoke
```

Comment:
```c
/*
   Notes:
    Attributes must be extracted from the same language as they were inserted
    in with 'MPI_Comm_set_attr'.  The notes for C and Fortran below explain
    why.
Notes for C:
    Even though the 'attr_value' argument is declared as 'void *', it is
    really the address of a void pointer.  See the rationale in the
    standard for more details.
*/
```

Reply:
You mean it was missing "notes for fortran"?

Reply:
Sorry for my unclear comment. `MPI_Comm_get_attr()` in `attr_api.txt` does not have this note while the original file does. Should we add this note in `attr_api.txt` like the other functions? Or is it intentionally omitted (e.g., because it will be autogenerated somehow)? If it is intentionally omitted, what is a criteria (and should it be part of this PR since it makes reviews more complicated)? If I shouldn't point it out because these notes and flags are added/removed for some certain reasons, I won't in the future review.

Reply:
Duplicated for `MPI_Comm_get_attr`, `MPI_Type_get_attr`, and `MPI_Win_get_attr` in #4955